iptables/src/utils.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2018 Trail of Bits, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <osquery/sdk.h>
+
+extern "C" {
+#include <libiptc/libiptc.h>
+}
+
+#include <netdb.h>
+#include <sys/socket.h>
+
+// Prepends a "!" to the given string if flag is present in the
+// given struct's invert flags.
+// NOTE(ww): Experimentally, it looks like recent versions of iptables
+// don't use these flags much -- they only seem to get set on the protocol
+// and a few other fields, with other fields receiving a mask instead.
+#define FLAGNEGATE(x, flag, str)                                               \
```

Comment:
nit, not sure if might want to prefer an inline function vs a macro here

Reply:
The problem here is polymorphism -- `x` can be a pointer to just about any `libip6?tc` structure, and `flag`'s width can vary. Is there a clean way to do that with inline functions in C++?

Reply:
ah, i see, i didn't read it closely enough -- not that i know. lgtm in that case